File: addons/resource_mail/models/resource_resource.py
# -*- coding: utf-8 -*-
# Part of Odoo. See LICENSE file for full copyright and licensing details.
from odoo import models
from odoo.addons.mail.tools.discuss import Store


class ResourceResource(models.Model):
    _inherit = 'resource.resource'

    def _store_avatar_card_fields(self, res: Store.FieldList):
        res.one("user_id", "_store_avatar_card_fields")
        res.extend(["name", "resource_type"])

File: addons/resource_mail/static/tests/mock_server/mock_models/resource_resource.js
import { ResourceResource } from "@resource/../tests/mock_server/mock_models/resource_resource";

import { patch } from "@web/core/utils/patch";

patch(ResourceResource.prototype, {
    _store_avatar_card_fields(res) {
        res.one("user_id", "_store_avatar_card_fields");
        res.extend(["name", "resource_type"]);
    },
});
